flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/casting/CastRulesTest.java

```diff
@@ -783,6 +824,45 @@
         return new GenericMapData(map);
     }
 
+    public static class MyStructuredType {
+        private long a;
+        private Long b;
+        private String c;
+        private String[] d;
+
+        public long getA() {
```

Comment:
if you want to keep your code concise you can also just have public mutable members. should be ok for testing code.